From 7b33106e3cf8a9d92fe4cc765e8842df010b16d6 Mon Sep 17 00:00:00 2001
From: Yong Zhao <Yong.Zhao@amd.com>
Date: Mon, 17 Apr 2017 15:34:26 -0400
Subject: [PATCH 2019/2094] drm/amdkfd: fix compiling warnings (specific for
 hybrid driver)

The below warnings were generated because of additional header files
used in the command-line for hybrid driver building:

/home/yozhao/git/compute/kernel/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c:23:0: warning: "pr_fmt" redefined [enabled by default]
 #define pr_fmt(fmt) "kfd2kgd: " fmt
 ^
In file included from /home/yozhao/git/compute/kernel/include/linux/kernel.h:13:0,
                 from /home/yozhao/git/compute/kernel/include/linux/list.h:8,
                 from /home/yozhao/git/compute/kernel/include/linux/agp_backend.h:33,
                 from /home/yozhao/git/compute/kernel/include/drm/drmP.h:35,
                 from /home/yozhao/git/compute/kernel/include/kcl/kcl_drm.h:6,
                 from /home/yozhao/git/compute/kernel/drivers/gpu/drm/amd/amdgpu/../backport/backport.h:4,
                 from <command-line>:0:
/home/yozhao/git/compute/kernel/include/linux/printk.h:261:0: note: this is the location of the previous definition
 #define pr_fmt(fmt) fmt
 ^

Fix the warnings by using #undef first.

Change-Id: Id52b8ac8c02d28eb4ebd1879af1a68bf6fb8b72c
Signed-off-by: Yong Zhao <Yong.Zhao@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c        | 1 +
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v7.c | 1 +
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c | 1 +
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c  | 1 +
 4 files changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
index b027adf..5a5fd55 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
@@ -20,6 +20,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#undef pr_fmt
 #define pr_fmt(fmt) "kfd2kgd: " fmt
 
 #include "amdgpu_amdkfd.h"
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v7.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v7.c
index 1b2b690..d2de59b 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v7.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v7.c
@@ -20,6 +20,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#undef pr_fmt
 #define pr_fmt(fmt) "kfd2kgd: " fmt
 
 #include <linux/fdtable.h>
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c
index 1503291..a8d36fb 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gfx_v8.c
@@ -20,6 +20,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#undef pr_fmt
 #define pr_fmt(fmt) "kfd2kgd: " fmt
 
 #include <linux/module.h>
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index ee3ae8f..6160111 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -20,6 +20,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#undef pr_fmt
 #define pr_fmt(fmt) "kfd2kgd: " fmt
 
 #include <linux/module.h>
-- 
2.7.4

